cdap-app-fabric/src/main/java/io/cdap/cdap/support/job/SupportBundleJob.java

```diff
@@ -0,0 +1,236 @@
+        String taskName = supportBundleState.getUuid().concat(": ").concat(className);
+        executeTask(supportBundleTask, basePath, className, taskName);
+      }
+      completeProcessing(basePath);
+    } catch (Exception e) {
+      LOG.warn("Can not execute the tasks ", e);
+    }
+  }
+
+  public void executeTask(
+    SupportBundleTask supportBundleTask,
+    String basePath,
+    String className,
+    String taskName) {
+    SupportBundleTaskStatus taskStatus = initializeTask(taskName, className);
+    executeTask(taskStatus, supportBundleTask, basePath, className, taskName, 0);
+  }
+
+  /**
+   * Execute each task to generate support bundle files
+   */
+  private void executeTask(
+    SupportBundleTaskStatus taskStatus,
+    SupportBundleTask supportBundleTask,
+    String basePath,
+    String className,
+    String taskName,
+    int retryCount) {
+    RunningTaskState runningTaskState = new RunningTaskState();
+    Future<SupportBundleTaskStatus> futureService =
+      executor.submit(
+        () -> {
+          try {
+            runningTaskState.setStartTime(System.currentTimeMillis());
+            updateTask(taskStatus, basePath, CollectionState.IN_PROGRESS);
+            supportBundleTask.collect();
+            taskStatus.setFinishTimestamp(System.currentTimeMillis());
+            updateTask(taskStatus, basePath, CollectionState.FINISHED);
+          } catch (Exception e) {
+            LOG.warn(
+              "Failed to execute task with supportBundleTask {} ", taskName,
+              e);
+            executeTaskAgainAfterFailed(supportBundleTask, className, taskName,
+                                        taskStatus, basePath, retryCount + 1);
+          }
+          return taskStatus;
+        });
+    runningTaskState.setFuture(futureService);
+    runningTaskStateQueue.offer(runningTaskState);
+  }
+
+  /**
+   * Execute all processing
+   */
+  public void completeProcessing(String basePath) {
+    while (!runningTaskStateQueue.isEmpty()) {
+      RunningTaskState runningTaskState = runningTaskStateQueue.poll();
+      SupportBundleTaskStatus supportBundleTaskStatus = null;
+      Future future = runningTaskState.getFuture();
+      try {
+        Long currentTime = System.currentTimeMillis();
+        Long futureStartTime = runningTaskState.getStartTime();
+        Long timeLeftBeforeTimeout =
+          TimeUnit.MINUTES.toMillis(maxThreadTimeout) - (currentTime - futureStartTime);
+        supportBundleTaskStatus = (SupportBundleTaskStatus) future.get(timeLeftBeforeTimeout,
+                                                                       TimeUnit.MILLISECONDS);
+        supportBundleTaskStatus.setFinishTimestamp(System.currentTimeMillis());
+        updateTask(supportBundleTaskStatus, basePath, CollectionState.FINISHED);
+      } catch (Exception e) {
+        LOG.error("The task for has failed or timeout more than five minutes ", e);
```

Comment:
Please catch different exceptions so that a more precise error message can be logged.